catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java

```diff
@@ -936,4 +951,17 @@ private void applyFilters(OpenSearchFilterVisitorObject visitor, WebClient clien
             client.replaceQueryParam(URL_SRC_PARAMETER, "");
         }
     }
+
+    private List<Metacard> processAdditionalForeignMarkups(Element element, String id) {
+        List<Metacard> metacards = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(markUpSet)) {
+            XMLOutputter xmlOutputter = new XMLOutputter();
+            if (markUpSet.contains(element.getName())) {
```

Comment:
✏️ It would be a bit more efficient and reduce complexity if you collapsed this condition with the wrapping condition. Then you wouldn't create the `XMLOutputter` unless you were going to use is.

`if (CollectionUtils.isNotEmpty(markUpSet) && markUpSet.contains(element.getName())`

Reply:
fixed